### ISSUE TITLE
How to detect the silence part in audio when using wavesurfer.js

### ISSUE THREAD

**Issue by truong24hdev** [katspaugh/wavesurfer.js#2632]
### The issue:

I am developing a feature to help users to highlight silences in an audio file, so I want to detect start time silent in audio files. I am developing it using ReactJs and wavesurfer-js library. I want to ask if it is possible for this library to assist me in implementing this feature. How can this library help me to detect when the wave has a lot of straight line (i.e. too much silence), so I can know when to start silent in audio.

![image](https://user-images.githubusercontent.com/118870130/208288322-e2a4858a-e70d-41ed-859c-859b77b2e963.png)
Look at the picture, you can see the silent part in the end of the audio, so I want to highlight this part

I've tried to check the docs if there is any function i can call or any property i can access on the WaveSurfer object but couldn't find any.

I want to detect start time silent in audio files. Could you tell me the way of doing this, please? thank you so much

**Comment by prakharpbuf** [comment]
I don't believe wavesurfer.js provides any methods for directly manipulating/analyzing audio. However, you can use the JavaScript Web Audio API for such features.
You can also use services like ffmpeg in your backend and have it return, for example, start time and end time of silences and do wavesurfer.addregion(options) or something to highlight/select the silence portion, depending on your use case.

**Comment by entonbiba** [comment]
@truong24hdev have a look at this section of the code, you can essentially add in there a part where it would check the peak and manipulate the progress and bar colours before it renders if it's 0. 

`setFillStyles`
https://github.com/katspaugh/wavesurfer.js/blob/a92da7f453a105bcf12ef00db9630770bd447db5/src/drawer.multicanvas.js#L566

**Comment by skysnolimit08** [comment]
You can ask WaveSurfer to give you an array of peaks in the waveform and then look for 0 values or sequences of 0 values within the array.  This is the line of code that I used to generate such an array.  I hope this is what you are looking for.  I don't remember why I said that accuracy is irrelevant.

	// length of output array/2, accuracy (irrelevant), don't popup a new window, start at 0,
	this.wavesurfer.exportPCM((this.totalDuration / 2) * 100, 10000, true, 0)

**Comment by prakharpbuf** [comment]
Do you still need help with this? I think I can help you.
I've been working a lot with the web audio API and audio manipulation. I think I can help you find the start and end times of silence portions in the audio file even if the peaks value is not completely 0. i.e. there is some noise/interference.
